### PR TITLE
Enhance the `bump_version.sh` script

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request does the following:
- Rename the `bump_version.sh` script to `bump-version`.
- Update the `bump-version` script to provide more functionality, including:
  - Bump the prerelease version when bumping the major, minor, or patch version. This allows you to bump from `0.0.1` to `0.1.0-rc.1` in one command for example.
  - Make the `git push` opt-in as an option.
  - Change the label used when bumping the build or prerelease versions. This allows you to generate a `0.0.1+dev.1` or a `0.1.0-alpha.1` with the script.
- Make the usage information more useful and add explicit flags to get help information.

## 💭 Motivation and context ##

I have wanted to update this script to remove some of the pain points I have with it such as having to run it twice to generate a prerelease of a new version. This seemed like as good a time as any to do so.

## 🧪 Testing ##

Automated tests pass. I used it to update the version in this pull request. I then used the prerelease version I published from this branch in my modified local [cisagov/cyhy_amis] and [cisagov/skeleton-packer] repositories and confirmed things work as expected from previously merged pull requests.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

[cisagov/cyhy_amis]: https://github.com/cisagov/cyhy_amis
[cisagov/skeleton-packer]: https://github.com/cisagov/skeleton-packer

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

File: bump-version [hunk 1]
```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+
+# bump_version.sh [--push] [--label LABEL] (major | minor | patch | prerelease | build | finalize | show)
```
Comment (dav3r) [inline bump-version]: Match new script name:
```suggestion
# bump-version [--push] [--label LABEL] (major | minor | patch | prerelease | build | finalize | show)
```

File: bump-version [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    esac
+  done
+fi
+
+if [ -n "$label" ] && [ "$with_prerelease" = false ] && [[ ! " ${commands_with_label[*]} " =~ [[:space:]]${bump_part}[[:space:]] ]]; then
+  invalid_option "Setting the label is only allowed for the following commands: ${commands_with_label[*]}"
+fi
+
+if [ "$with_prerelease" = true ] && [[ ! " ${commands_with_prerelease[*]} " =~ [[:space:]]${bump_part}[[:space:]] ]]; then
+  invalid_option "Changing the prerelease is only allowed in conjunction with the following commands: ${commands_with_prerelease[*]}"
```
Comment (dav3r) [inline bump-version:117]: Make error messages consistent:
```suggestion
  invalid_option "Changing the prerelease is only allowed for the following commands: ${commands_with_prerelease[*]}"
```

If you prefer one error message grammar over the other (e.g. "in conjunction with" vs. "for"), feel free to use that, just make sure both error messages are consistent.

**Review by dav3r — APPROVED** [review]
Love all of these quality of life improvements!  Thanks for doing these. 👍 👍 👍

File: bump-version (hunk 2 above)
Comment (mcdonnnj) [reply bump-version:117]: Hm. The main issue (to me) is it's two different kinds of errors. The one you marked is allowed to exist on its own, or with a subset of commands. The other can only be used with a subset of commands.

File: bump-version (hunk 1 above)
Comment (mcdonnnj) [reply bump-version]: Ack missed that one.

**Review by dv4harr10 — APPROVED** [review]

File: bump-version (hunk 2 above)
Comment (dav3r) [reply bump-version:117]: OK, that is fair.  I withdraw my suggestion.

**Pull request merged**